src/Libraries/DynamoUnits/Utilities.cs

```diff
@@ -404,5 +404,129 @@ internal static bool TryParseTypeId(string typeId, out string typeName, out Vers
+            // support becomes available, we can extend this to work on other platforms.
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            try
+            {
+                // Use reflection to dynamically load DynamoInstallDetective at runtime.
+                // This avoids the need for a direct project reference and InternalsVisibleTo,
+                // maintaining cross-platform compatibility for the DynamoUnits library.
+                var dynamoInstallDetectiveAssembly = Assembly.LoadFrom(
+                    Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), 
+                    "DynamoInstallDetective.dll"));
+
+                var ascWrapperType = dynamoInstallDetectiveAssembly.GetType("DynamoInstallDetective.AscSdkWrapper");
+                if (ascWrapperType == null)
+                {
+                    return; // AscSdkWrapper type not found
+                }
+
+                // Get major versions using reflection: AscSdkWrapper.GetMajorVersions()
+                var getMajorVersionsMethod = ascWrapperType.GetMethod("GetMajorVersions", 
+                    BindingFlags.Public | BindingFlags.Static);
+                var majorVersions = (string[])getMajorVersionsMethod?.Invoke(null, null);
+
+                if (majorVersions == null) return;
+
+                // Get the ASC_STATUS enum for comparison
+                var ascStatusType = ascWrapperType.GetNestedType("ASC_STATUS");
+                var successValue = Enum.Parse(ascStatusType, "SUCCESS");
+
+                foreach (var majorVersion in majorVersions)
+                {
+                    // Create AscSdkWrapper instance: new AscSdkWrapper(majorVersion)
+                    var ascWrapper = Activator.CreateInstance(ascWrapperType, majorVersion);
+
+                    // Call GetInstalledPath using reflection
+                    var getInstalledPathMethod = ascWrapperType.GetMethod("GetInstalledPath");
+                    var parameters = new object[] { string.Empty };
+                    var result = getInstalledPathMethod?.Invoke(ascWrapper, parameters);
+
+                    // Check if result equals ASC_STATUS.SUCCESS
+                    if (result != null && result.Equals(successValue))
+                    {
+                        var installPath = (string)parameters[0];
+                        var schemaPath = Path.Combine(installPath, "coreschemas", "unit");
+                        candidateDirectories.Add(schemaPath);
+                    }
+                }
+            }
+            catch
+            {
+                // Ignore errors when discovering ASC paths - this is optional discovery.
+                // DynamoInstallDetective.dll might not be available on some deployments,
+                // or ASC might not be installed on the system.
+            }
+        }
+
```

Comment:
`ForgeUnits.SchemaUtility.addDefinitionsFromFolder` will succeed even if the specified `schemaDirectory` is empty. Therefore, it’s important to verify that required subdirectories such as `quantity`, `symbol`, and others exist under the `unit` directory; otherwise, the directory won’t qualify as a valid source for `ForgeUnits.UnitsEngine`.